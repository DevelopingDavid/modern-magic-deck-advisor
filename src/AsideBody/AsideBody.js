import React, { Component } from 'react';
// import Controls from '../Controls/Controls.js';
import '../styles/main.scss';
import CardList from '../CardList/CardList.js';
import FaveList from '../FaveList/FaveList.js';
import WishList from '../WishList/WishList.js';

function AsideBody(props) {
  if (props.asideView === "myCardList") {
    return (
      <div>
        <CardList 
          cardNames={props.cardNames} 
          removeListItem={props.removeListItem}
          setCardCount={props.setCardCount}
          userCardsData={props.userCardsData} 
          compareBuilds={props.compareBuilds}/>
      </div>
    )
  } else if (props.asideView === "faveDecks") {
    return (
      <div>
        <FaveList 
          userFaveDecks={props.userFaveDecks}
          removeFaveListItem={props.removeFaveListItem}/>
      </div>
    )
  } else if (props.asideView === "wishList"){
    return (
      <div>
        <WishList 
          wishList={props.wishList}
          removeWishListItem={props.removeWishListItem}/>
      </div>
    )
  }
}


export default AsideBody;