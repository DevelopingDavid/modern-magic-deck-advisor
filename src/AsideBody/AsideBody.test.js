import React from 'react';
import AsideBody from './AsideBody';
import { shallow } from 'enzyme';

const cardNames = ['blood moon', 'breeding pool'];
const userCards = [{ cardName: 'blood moon', cardCount: 1 }, { cardName: 'breeding pool', cardCount: 1 }];

const removeListItemMock = jest.fn();
const setCardCountMock = jest.fn();

describe('AsideBody', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <AsideBody 
        cardNames={cardNames} 
        removeListItem={removeListItemMock}
        setCardCount={setCardCountMock}
        userCards={userCards} 
      />
    )
  });
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
});

