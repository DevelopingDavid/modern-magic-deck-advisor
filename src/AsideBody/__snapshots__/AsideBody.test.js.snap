// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsideBody should match the snapshot 1`] = `
<div>
  <CardList
    cardNames={
      Array [
        "blood moon",
        "breeding pool",
      ]
    }
    removeListItem={[MockFunction]}
    setCardCount={[MockFunction]}
  />
</div>
`;
